services/system/StagedTx/plugin/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+#[allow(warnings)]
+mod bindings;
+
+use bindings::accounts::plugin::api::get_account;
+use bindings::exports::staged_tx::plugin::{
+    proposer::Guest as Proposer, respondent::Guest as Respondent,
+};
+use bindings::exports::transact_hook_tx_transform::{Guest as HookTxTransform, *};
+use bindings::host::common::{client as Client, server as Server, types::Error};
+use bindings::transact::plugin::{hooks::hook_tx_transform_label, intf::add_action_to_transaction};
+use psibase::fracpack::Pack;
+use psibase::services::staged_tx::action_structs::propose;
+use psibase::{AccountNumber, Checksum256, Hex, MethodNumber};
+use staged_tx::action_structs::*;
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+mod errors;
+use errors::ErrorType;
+
+fn validate_account(account: &str) -> Result<(), Error> {
+    match get_account(account) {
+        Ok(Some(_)) => Ok(()),
+        Ok(None) => Err(ErrorType::InvalidAccount(account.to_string()).into()),
+        Err(e) => Err(e),
+    }
+}
+
+fn get_assert_caller(context: &str, allowed_apps: &[&str]) -> Result<(), Error> {
+    let caller = Client::get_sender_app();
+    let caller_app = caller.app.as_ref().unwrap();
+    if caller.app.is_none() || !allowed_apps.contains(&caller_app.as_str()) {
+        return Err(ErrorType::Unauthorized(context.to_string(), caller_app.to_string()).into());
+    }
+    Ok(())
+}
+
+#[derive(Deserialize, Serialize)]
+struct StagedTxDetails {
+    data: StagedTxData,
+}
+
+#[derive(Deserialize, Serialize)]
+struct StagedTxData {
+    details: StagedTxDetailsInner,
+}
+
+#[derive(Deserialize, Serialize)]
+struct StagedTxDetailsInner {
+    txid: String,
+    #[serde(rename = "proposeBlock")]
```

Comment:
rename_all = "camelCase" is less repetitive.

Reply:
done